source/extensions/tracers/zipkin/tracer.cc

```diff
@@ -66,14 +68,25 @@ SpanPtr Tracer::startSpan(const Tracing::Config& config, const std::string& span
   span_ptr->setName(span_name);
 
   // Set the span's kind (client or server)
-  if (config.operationName() == Tracing::OperationName::Egress) {
-    annotation.setValue(CLIENT_SEND);
+  if (split_spans_for_request_) {
+    // If previous context is inner context then this span must be span created for upstream
+    // request. Client span will be created for upstream request when split_spans_for_request is
+    // set to true.
```

Comment:
```suggestion
    // If the previous context is an inner context then this span must be a span created for an upstream
    // request. A client span will be created for an upstream request when split_spans_for_request is
    // set to true.
```

This clause is inside the `if (split_spans_for_request_)` part, so it is a bit unclear when reading "Client span will be created for upstream request when split_spans_for_request is set to true", because it is already set to true.

Reply:
Got it. Thanks. :)